nixos/modules/hardware/video/nvidia.nix

```diff
@@ -180,6 +189,25 @@ in
     environment.systemPackages = [ nvidia_x11.bin nvidia_x11.settings ]
       ++ lib.filter (p: p != null) [ nvidia_x11.persistenced ];
 
+    systemd.services = let
+      nvidiaService = type: {
+        description = "NVIDIA system ${type} actions";
+        before = [ "systemd-${type}.service" ];
+
+        path = with pkgs; [ coreutils nvidia_x11.bin ];
+        serviceConfig = {
+          Type = "oneshot";
+        };
+
+        script = ''
+          logger -t ${type} -s "nvidia-${type}.service"
+          nvidia-sleep.sh "${type}"
+        '';
+
+        requiredBy = [ "systemd-${type}.service" ];
+      };
+    in builtins.listToAttrs (map (t: nameValuePair "nvidia-${t}" (nvidiaService t)) ["hibernate" "resume" "suspend"]);
```

Comment:
Reading through `man systemd-suspend.service`, I don't really get how this should work with `systemd-resume.service`, and wonder if `hybrid-sleep` and `suspend-then-hibernate` are missing.

Would it be possible to add this to `/usr/lib/systemd/system-sleep/`? I assume nvidia also doesn't support https://www.freedesktop.org/wiki/Software/systemd/inhibit/, do they?

Reply:
They have both the script located in `lib/systemd/system-sleep/nvidia` and the service. As for the service for resuming, I messed up the `before` and `requiredBy` attributes.